client/src/components/mapping/InputColumn/index.tsx

```diff
@@ -124,55 +88,78 @@ const InputColumn = ({ input }: Props) => {
-                        <Tag intent={'success'} large={true}>
-                          {input.sqlValue.table}
-                        </Tag>
-                      </div>
-                    )
-                  },
-                  {
-                    text: (
-                      <div className="stacked-tags">
-                        <Tag minimal={true}>COLUMN</Tag>
-                        <Tag intent={'success'} large={true}>
-                          {input.sqlValue.column}
-                        </Tag>
-                      </div>
-                    )
+          <div className="sql-input-form">
+            <ColumnSelect
+              tableChangeCallback={(table: string) => {
+                updateInput({
+                  variables: {
+                    inputId: input.id,
+                    data: { table, column: '' }
+                  }
+                });
+              }}
+              columnChangeCallback={(column: string) => {
+                updateInput({
+                  variables: {
+                    inputId: input.id,
+                    data: { column }
                   }
-                ]}
-              />
-              <div className="stacked-tags" onClick={e => e.stopPropagation()}>
-                <Tag>SCRIPT</Tag>
+                });
+              }}
+              joinChangeCallback={(joinId: string, newjoin: Join): void => {
+                updateJoin({
+                  variables: {
+                    joinId,
+                    data: newjoin
+                  }
+                });
+              }}
+              addJoinCallback={(newjoin: Join): void => {
+                addJoin({
+                  variables: {
+                    columnId: input.sqlValue.id,
+                    join: newjoin
+                  }
+                });
+              }}
+              deleteJoinCallback={(joinId: string): void => {
+                deleteJoin({
+                  variables: {
+                    joinId
+                  }
+                });
+              }}
+              initialTable={input.sqlValue.table}
+              initialColumn={input.sqlValue.column}
+              initialJoins={input.sqlValue.joins}
+              sourceSchema={source.credential.schema as ISourceSchema}
+              primaryKeyTable={resource.primaryKeyTable}
+              popoverProps={{
+                autoFocus: true,
+                boundary: 'viewport',
+                canEscapeKeyClose: true,
+                lazy: true,
+                position: Position.TOP,
+                usePortal: true
+              }}
+            />
+            <div className="sql-input-form-script">
+              <div className="stacked-tags">
+                <Tag minimal={true}>SCRIPT</Tag>
                 <ScriptSelect
-                  loading={loadUpdInput}
-                  selectedScript={input.script}
+                  selectedScript={input.script || ''}
```

Comment:
is the default empty string really necessary ?

Reply:
It's not, indeed 👍